Source/Core/Core/PowerPC/BreakPoints.cpp

```diff
@@ -10,12 +10,57 @@
 #include <string>
 #include <vector>
 
+#include <expr.h>
+
 #include "Common/CommonTypes.h"
 #include "Common/DebugInterface.h"
 #include "Common/Logging/Log.h"
 #include "Core/Core.h"
 #include "Core/PowerPC/JitInterface.h"
 #include "Core/PowerPC/MMU.h"
+#include "Core/PowerPC/PowerPC.h"
+
+static bool ParseGPR(const char* name, int* index)
+{
+  if (name[0] == 'r' && isdigit(name[1]) &&
+      (name[2] == '\0' || isdigit(name[2]) && (name[3] == '\0')))
+  {
+    int idx = name[1] - '0';
+    if (name[2] != '\0')
+    {
+      idx = idx * 10 + (name[2] - '0');
+    }
+    if (idx < 32)
+    {
+      *index = idx;
+      return true;
+    }
+  }
+
+  *index = 0;
+  return false;
+}
+
+static bool EvaluateExpression(const std::string& expression)
+{
+  static expr_var_list vars = {0};
+  expr* e = expr_create(expression.c_str(), expression.length(), &vars, nullptr);
+  if (!e)
+    return false;
+
+  for (auto* v = vars.head; v != nullptr; v = v->next)
+  {
+    int index = 0;
+    if (ParseGPR(v->name, &index))
+      v->value = GPR(index);
```

Comment:
Should probably return false if ParseGPR fails

Reply:
This is actually intentional (for now). This allows expressions to introduce their own variables. If that feature is cut, then yes, we would want to fail here.